packages/emotion-theming/types/index.d.ts

```diff
@@ -3,30 +3,43 @@
 
 import * as React from 'react'
 
-import { AddOptionalTo, PropsOf } from './helper'
+import {
+  StyledComponent,
+  StyledOptions,
+  CreateStyledComponent,
+  StyledTags
+} from '@emotion/styled'
+import { PropsOf, Omit } from '@emotion/styled-base'
 
 export interface ThemeProviderProps<Theme> {
   theme: Partial<Theme> | ((outerTheme: Theme) => Theme)
   children?: React.ReactNode
 }
 
-export function ThemeProvider<Theme>(
-  props: ThemeProviderProps<Theme>
-): React.ReactElement
+export type ThemeProvider<Theme = any> = <T extends Theme = Theme>(
+  props: ThemeProviderProps<T>
+) => React.ReactElement
 
-export function useTheme<Theme>(): Theme
-
-/**
- * @todo Add more constraint to C so that
- * this function only accepts components with theme props.
- */
-export function withTheme<C extends React.ComponentType<any>>(
+export type withTheme<Theme = any> = <
```

Comment:
Change log:

withTheme can now have the Theme type specified when calling it. For example `withTheme<MyTheme>(MyComponent)`

Breaking change, if you were specifying the ComponentType you will need to remove the generic parameter. Recommend following example setup in the TypeScript docs under theming section